aten/src/ATen/core/ivalue.cpp

```diff
@@ -101,6 +101,46 @@ TypePtr IValue::type() const {
   TORCH_INTERNAL_ASSERT(false, "unhandled case in IValue::type()");
 }
 
+void IValue::visit(const std::function<bool (const IValue &)>& visitor) const {
```

Comment:
I recommend against adding this to IValue: it does not contain tests, and is not even turned on in this PR. It is the kind of simple function that people looking at ivalue will think it tested and implemented correctly. This should be implemented locally where needed unless we commit to completely implementing it and testing it.